WaveManager/src/main/java/com/github/sef24sp4/wavemanager/WaveManager.java

```diff
@@ -0,0 +1,201 @@
+package com.github.sef24sp4.wavemanager;
+
+import com.github.sef24sp4.common.enemy.CommonEnemy;
+import com.github.sef24sp4.common.enemy.EnemyRole;
+import com.github.sef24sp4.common.enemy.EnemySPI;
+import com.github.sef24sp4.common.entities.IEntity;
+import com.github.sef24sp4.common.interfaces.IGameSettings;
+import com.github.sef24sp4.common.vector.Coordinates;
+
+import java.util.*;
+
+public class WaveManager implements IWaveManager {
+
+	private Collection<IEntity> enemyComposition = new HashSet<>();
+
+	/**
+	 * used instead of a map.
+	 */
+	private final IGameSettings gameSettings;
+
+	private int waveNumber;
+
+	private long waveDifficulty;
+
+	private final int timeUntilNextWaveInSeconds;
+
+	private long timeOfLastCheck;
+
+	private WaveStatus waveStatus = WaveStatus.ONGOING;
+
+	private Map<EnemyRole, List<EnemySPI>> enemyCatalog = new HashMap<>();
+
+	public WaveManager(int startWaveNumber, int waveCountdownInSeconds, IGameSettings gameSettings) {
```

Comment:
Might be relevant to make JavaDoc, which specifies valid values for `startWaveNumber`.

When not all values of a type is legal, then JavaDoc should specify when a value is or is not legal input.